backend/src/modules/courses/course.repository.ts

```diff
@@ -35,4 +41,32 @@ export class CourseRepository {
       throw new BadRequestException(`Error fetching course: ${error.message}`);
     }
   }
+
+  async updateOne(id: string, updateData: Partial<Course>): Promise<void> {
+    try {
+      await this.courseRepository.update(id, updateData);
+    } catch (error) {
+      throw new BadRequestException(`Failed to update course: ${error.message}`);
+    }
+  }
+
+  async teacherExists(teacherId: string): Promise<boolean> {
+    try {
+      const teacher = await this.userRepository.findOne({ where: { id: teacherId } });
```

Comment:
I think user repository not related to course repository that's independent parts.